### PR TITLE
ISDK-2798: Enable treat compiler errors as warnings

### PR DESCRIPTION
**Pull request by paynerc**
## Description

With the `3.1.0+` releases of the TwilioVideo.framework iOS SDK we had an issue where a Swift `NS_SWIFT_NAME` mapping was failing to work properly and was causing issues for customers who have their `Swift Compiler - Treat Warnings as Errors` build setting set to true [#72](https://github.com/twilio/twilio-video-ios/issues/72). We finally tracked this issue down, but it would be helpful if we were able to catch this sort of issue during our own internal development process.

With this PR, I have enabled both the `Apple Clang - Treat Warnings as Errors` and the `Swift Compiler - Treat Warnings as Errors` build settings. This of course caused our CI process to fail. Then I added the necessary CI configuration to pull from our internal spec repository where we have access to prerelease versions and updated the `Podfile` to consume `3.2.2-rc1`. Once this was done, the build was successful again.

## Breakdown

- Enabled both the `Apple Clang - Treat Warnings as Errors` and the `Swift Compiler - Treat Warnings as Errors` build settings for all targets in the project.
- Added the `.circleci/.netrc` template file for allowing our CI VMs the ability to consume internal artifacts
- Added the `.circleci/setup_internal_pod_repo.sh` script to set up the necessary items for consuming internal artifacts and configuring the internal spec repo.
- Updated `.circleci/config.yml` to invoke `.circleci/setup_internal_pod_repo.sh` during the setup phase.
- Updated the `Podfile` to consume TwilioVideo `3.2.2-rc1`

## Validation

- Ran a series of CI jobs along the way:
  - 1. Enabled the project settings and validated that the CI build failed due to the issue reported in [#72](https://github.com/twilio/twilio-video-ios/issues/72)
  - 2. Configured the CI environment with the ability to consume internal pod artifacts, updated the `Podfile` to consume an artifact which fixes the issue and validated that the build now passes.

## Additional Notes

~Once `3.2.2` is officially released I will modify the `Podfile` back so it consumes a publicly available release before we merge this to `master`.~

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -1,13 +1,16 @@
+source 'git@github.com:twilio/cocoapod-specs-internal.git'
+source 'https://cdn.cocoapods.org/'
```
[the remark this answers is not in the document]
Comment (paynerc) [reply Podfile]: Yes, I was thinking the same thing.

File: .circleci/setup_internal_pod_repo.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+chmod 600 .circleci/.netrc
+cp .circleci/.netrc ~/.netrc
+sed -i '' "s/BINTRAY_USERNAME/${BINTRAY_USERNAME}/g" ~/.netrc
+sed -i '' "s/BINTRAY_API_KEY/${BINTRAY_API_KEY}/g" ~/.netrc
```
[the remark this answers is not in the document]
Comment (paynerc) [reply .circleci/setup_internal_pod_repo.sh]: Well, we have been trying to move away from `secrets.tar` on the SDK side of things. In fact at this point I have it all removed except for the provisioning profiles. We have numerous profiles for all of the various testing configurations and such and with that I was unable to store the lot of them in the environment variables inside of CircleCI. But anything that is credential based is now in Env Variables in the SDK project on CI. This way, if for some reason we need to revoke access to a certain set of crews, we no longer have to modify code in the repo. Just change the env vars and re run the job.

[the remark this answers is not in the document]
Comment (timrozum) [reply .circleci/setup_internal_pod_repo.sh]: We talked offline and I think this is good now.

**Comment by paynerc** [comment]
@timrozum I addressed "reverting" the `Podfile` back to official release status. Can you give it one more set of 👀 before we make it so?

File: README.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # Twilio iOS Video App
 
+![CircleCI](https://circleci.com/gh/twilio/twilio-video-app-ios.svg?style=shield&circle-token=5472df4715fce13ee02276b13b5325acd40128b4)
```
Comment (timrozum) [inline README.md]: Cool!

**Comment by timrozum** [comment]
Looks good!

**Pull request merged**